packages/dd-trace/src/appsec/rule_manager.js

```diff
@@ -26,137 +20,66 @@ function loadRules (config) {
 }
 
 function updateWafFromRC ({ toUnapply, toApply, toModify }) {
-  const batch = new Set()
-
-  const newRulesData = new SpyMap(appliedRulesData)
-  let newRuleset
-  let newRulesetId
-  const newRulesOverride = new SpyMap(appliedRulesOverride)
-  const newExclusions = new SpyMap(appliedExclusions)
-  const newCustomRules = new SpyMap(appliedCustomRules)
   const newActions = new SpyMap(appliedActions)
 
+  let wafUpdated = false
+  let wafUpdatedSuccess = true
+
   for (const item of toUnapply) {
-    const { product, id } = item
+    if (!['ASM_DD', 'ASM_DATA', 'ASM'].includes(item.product)) continue
 
-    if (product === 'ASM_DATA') {
-      newRulesData.delete(id)
-    } else if (product === 'ASM_DD') {
-      if (appliedRulesetId === id) {
-        newRuleset = defaultRules
+    try {
+      waf.wafManager.remove(item.path)
+      item.apply_state = ACKNOWLEDGED
+      wafUpdated = true
+
+      if (item.product === 'ASM') {
+        newActions.delete(item.id)
       }
-    } else if (product === 'ASM') {
-      newRulesOverride.delete(id)
-      newExclusions.delete(id)
-      newCustomRules.delete(id)
-      newActions.delete(id)
+    } catch (e) {
+      wafUpdatedSuccess = false
+      item.apply_state = ERROR
+      item.apply_error = e.toString()
```

Comment:
Is it intentional to only care about the error type and message? Calling `.toString()` is pretty bad for Node.js errors such that the stack and other properties are all lost. In case the error should be inspected, I would just use `util.inspect(error)`.

The same for other spots.

Reply:
Yes, it is. Tracer only reports error type and message in `apply_error` for RC items. 
- https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/remote_config/manager.js#L299
- https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/remote_config/manager.js#L314
- https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/remote_config/manager.js#L323

Reply:
I'll let @simon-id answer this :)

Reply:
Ye I coded it so back then so it only gets the type and message, but honestly we've never had to dig in the backend to use these errors, i don't think they're even showed to the customers. One day if this become important we can include the stack and all. For now I'd say we can keep it like this

Reply:
Errors are actually very expensive, so if the stack is not needed, I wonder if we should even use errors in that case.

Reply:
these errors should never happen anyway, this is a cold path, it executes every 5 seconds, and the chance of having an exception is low. Is it really worth to make that optimization in this PR ? I'd say meh. Carles is off for the week anyway. Something to keep in mind for a future big optimization PR I'd say